### ISSUE TITLE
Plugin verifier report duplicate plugin descriptors when combined JAR is used

### ISSUE THREAD

**Issue by jansorg** [JetBrains/intellij-platform-gradle-plugin#1673]
### What happened?

With the new `pluginModule(implementation(project(":plugin-java")))` syntax, the plugin verifier reports an error about duplicate jars.

### Relevant log output or stack trace

```shell
The configuration file 'appmap-java.xml' for optional dependency 'com.intellij.java' failed to be resolved: Found multiple plugin descriptors 'appmap-java.xml' from 'intellij-appmap-0.67.0.jar' and 'appmap-java.xml' from 'plugin-java-base.jar'.
```

### Steps to reproduce

Add the dependency listed above.
Further discussion at https://jetbrains-platform.slack.com/archives/C05C80200LS/p1719326254555919

### Gradle IntelliJ Plugin version

2.0.0.beta7

### Gradle version

8.6

### Operating System

Linux

**Comment by jansorg** [comment]
I'm not sure if the plugin verifier is the culprit here.
The verifier seems to receive two JARs: once the composed jar and once the jar which was used to create the composed jar.

**Comment by hsz** [comment]
Fixed with [640e356](https://github.com/JetBrains/intellij-platform-gradle-plugin/commit/640e3568d60d540b53a51dc903f73638ba865637)
